### PR TITLE
nodejs_18: 18.17.1 -> 18.18.0

### PR DESCRIPTION
**Pull request by screendriver**
## Description of changes

https://nodejs.org/en/blog/release/v18.18.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/web/nodejs/v18.nix [hunk 1]
```diff
@@ -8,7 +8,7 @@ let
 in
 buildNodejs {
   inherit enableNpm;
-  version = "18.17.1";
+  version = "18.18.0";
   sha256 = "sha256-8hXPA9DwDwesC2dMaBn4BMFULhbxUtoEmAAirsz15lo=";
```
Comment (cornedor) [inline pkgs/development/web/nodejs/v18.nix:12]: The hash need an update

```
error: hash mismatch in fixed-output derivation '/nix/store/8vsl6zffzjl1jw9b8agsrvvdil00lf94-node-v18.18.0.tar.xz.drv':
         specified: sha256-8hXPA9DwDwesC2dMaBn4BMFULhbxUtoEmAAirsz15lo=
            got:    sha256-5NTbrDY02Z+JLwDbR9p4+YSTwzlYLoqV+y3Vn1z+D5A=
```

Comment (screendriver) [reply pkgs/development/web/nodejs/v18.nix]: True. Thanks for the hint 👍

**Comment by aduh95** [comment]
What is needed to move this forward?

**Comment by lheckemann** [comment]
Thanks for your contribution! It seems like #256139 was already merged by the time this was opened though, sorry for the duplicated effort!

**Pull request closed without merging**